cairo/crates/pragma_dispatcher/src/contract.cairo

```diff
@@ -0,0 +1,255 @@
+            self.call_get_all_feeds()
+        }
+
+        /// Dispatch updates through the Hyperlane mailbox for the specified list
+        /// of [Span<FeedId>].
+        fn dispatch(self: @ContractState, feed_ids: Span<FeedId>) -> HyperlaneMessageId {
+            Default::default()
+        }
+    }
+
+    // ================== COMPONENTS IMPLEMENTATIONS ==================
+
+    #[abi(embed_v0)]
+    impl UpgradeableImpl of IUpgradeable<ContractState> {
+        fn upgrade(ref self: ContractState, new_class_hash: ClassHash) {
+            self.ownable.assert_only_owner();
+            self.upgradeable.upgrade(new_class_hash);
+        }
+    }
+
+    // ================== PRIVATE CALLERS OF SUB CONTRACTS ==================
+
+    impl PragmaFeedsRegistryWrapper of IPragmaFeedsRegistryWrapper<ContractState> {
+        /// Calls feed_exists from the Pragma Feeds Registry contract.
+        fn call_feed_exists(self: @ContractState, feed_id: FeedId) -> bool {
+            let registry_dispatcher = IPragmaFeedRegistryDispatcher {
+                contract_address: self.pragma_feed_registry_address.read()
+            };
+            registry_dispatcher.feed_exists(feed_id)
+        }
+
+        /// Calls get_all_feeds from the Pragma Feeds Registry contract.
+        fn call_get_all_feeds(self: @ContractState) -> Array<FeedId> {
+            let registry_dispatcher = IPragmaFeedRegistryDispatcher {
+                contract_address: self.pragma_feed_registry_address.read()
+            };
+            registry_dispatcher.get_all_feeds()
+        }
+    }
+
+    impl HyperlaneMailboxWrapper of IHyperlaneMailboxWrapper<ContractState> {
+        /// Calls dispatch from the Hyperlane Mailbox contract.
+        fn call_dispatch(self: @ContractState, message_body: Bytes) -> HyperlaneMessageId {
+            let mailbox_dispatcher = IMailboxDispatcher {
+                contract_address: self.hyperlane_mailbox_address.read()
+            };
+
+            mailbox_dispatcher
+                .dispatch(
+                    0_u32, // destination_domain
+                    0_u256, // recipient_address
+                    message_body,
+                    0_u256, // fee_amount
+                    Option::<Bytes>::None(()), // _custom_hook_metadata
+                    Option::<ContractAddress>::None(()), // _custom_hook
+                )
+        }
+    }
+
+    impl PragmaOracleWrapper of IPragmaOracleWrapper<ContractState> {
+        /// Calls get_data from the Pragma Oracle contract.
+        fn call_get_data(
+            self: @ContractState, data_type: DataType, aggregation_mode: AggregationMode,
+        ) -> PragmaPricesResponse {
+            let pragma_dispatcher = IPragmaABIDispatcher {
+                contract_address: self.pragma_oracle_address.read()
+            };
+
+            pragma_dispatcher.get_data(data_type, aggregation_mode)
+        }
+    }
+
+    impl SummaryStatsWrapper of ISummaryStatsWrapper<ContractState> {
```

Comment:
there's a bit of overhead to these wrapper impl tho but that's something we can keep in mind when optimizing